webgpu.h

```diff
@@ -1373,891 +1442,2452 @@ typedef struct WGPURequestDeviceCallbackInfo {
     WGPU_NULLABLE void* userdata2;
 } WGPURequestDeviceCallbackInfo WGPU_STRUCTURE_ATTRIBUTE;
 
+#define WGPU_REQUEST_DEVICE_CALLBACK_INFO_INIT _wgpu_MAKE_INIT_STRUCT(WGPURequestDeviceCallbackInfo, { \
+    /*.nextInChain=*/NULL _wgpu_COMMA \
+    /*.mode=*/WGPUCallbackMode_WaitAnyOnly _wgpu_COMMA \
+    /*.callback=*/NULL _wgpu_COMMA \
+    /*.userdata1=*/NULL _wgpu_COMMA \
+    /*.userdata2=*/NULL _wgpu_COMMA \
+})
+
 typedef struct WGPUUncapturedErrorCallbackInfo {
     WGPUChainedStruct const * nextInChain;
     WGPUUncapturedErrorCallback callback;
     WGPU_NULLABLE void* userdata1;
     WGPU_NULLABLE void* userdata2;
 } WGPUUncapturedErrorCallbackInfo WGPU_STRUCTURE_ATTRIBUTE;
 
+#define WGPU_UNCAPTURED_ERROR_CALLBACK_INFO_INIT _wgpu_MAKE_INIT_STRUCT(WGPUUncapturedErrorCallbackInfo, { \
+    /*.nextInChain=*/NULL _wgpu_COMMA \
+    /*.callback=*/NULL _wgpu_COMMA \
+    /*.userdata1=*/NULL _wgpu_COMMA \
+    /*.userdata2=*/NULL _wgpu_COMMA \
+})
+
 /** @} */
 
+/**
+ * Default values can be set using WGPU_ADAPTER_INFO_INIT as initializer.
+ */
 typedef struct WGPUAdapterInfo {
     WGPUChainedStructOut * nextInChain;
     /**
      * This is an \ref OutputString.
+     *
+     * Defaults to @ref WGPU_STRING_VIEW_INIT
      */
     WGPUStringView vendor;
     /**
      * This is an \ref OutputString.
+     *
+     * Defaults to @ref WGPU_STRING_VIEW_INIT
      */
     WGPUStringView architecture;
     /**
      * This is an \ref OutputString.
+     *
+     * Defaults to @ref WGPU_STRING_VIEW_INIT
      */
     WGPUStringView device;
     /**
      * This is an \ref OutputString.
+     *
+     * Defaults to @ref WGPU_STRING_VIEW_INIT
      */
     WGPUStringView description;
+    /**
+     * Defaults to @ref WGPUBackendType_Undefined
+     */
     WGPUBackendType backendType;
+    /**
+     * Defaults to @ref WGPUAdapterType_Unknown
+     */
     WGPUAdapterType adapterType;
+    /**
+     * Defaults to 0
+     */
     uint32_t vendorID;
+    /**
+     * Defaults to 0
+     */
     uint32_t deviceID;
 } WGPUAdapterInfo WGPU_STRUCTURE_ATTRIBUTE;
 
+#define WGPU_ADAPTER_INFO_INIT _wgpu_MAKE_INIT_STRUCT(WGPUAdapterInfo, { \
+    /*.nextInChain=*/NULL _wgpu_COMMA \
+    /*.vendor=*/WGPU_STRING_VIEW_INIT _wgpu_COMMA \
+    /*.architecture=*/WGPU_STRING_VIEW_INIT _wgpu_COMMA \
+    /*.device=*/WGPU_STRING_VIEW_INIT _wgpu_COMMA \
+    /*.description=*/WGPU_STRING_VIEW_INIT _wgpu_COMMA \
+    /*.backendType=*/WGPUBackendType_Undefined _wgpu_COMMA \
+    /*.adapterType=*/WGPUAdapterType_Unknown _wgpu_COMMA \
```

Comment:
I think this should default to a zero value. `Unknown` is a value that comes back from `GetInfo` and tells you the adapter type is unknown. (Though it could reasonably be argued that zero and `Unknown` should be the same value.)

That said I'm also not really sure `WGPUAdapterInfo` should even _have_ an `INIT` macro (since it's output-only). Dawn has it but maybe we should remove it? (I will deal with this in a followup.)

Reply:
Thing is, there is no zero-value for type `WGPUAdapterType`, and forcing a value of 0 has not been a success in my attempts above

Reply:
Changed this to `_wgpu_ENUM_ZERO_INIT(WGPUAdapterType)`